proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -0,0 +1,1117 @@
+    LOGGER.trace("Entering ampIfDefArg()");
+
+    loop : for (;;) {
+      if (currChar == ')') {
+        break loop;
+      }
+      // Watch for comments.
+      if (currChar == '/') {
+        getChar();
+        if (currChar != '*') {
+          append('/');
+          continue loop;
+        } else {
+          String s = currText.toString();
+          comment();
+          currText.replace(0, currText.length(), s);
+          continue loop;
+        }
+      }
+      append();
+      getChar();
+    }
+    return makeToken(PreprocessorParser.ID);
+  }
+
+  ProToken colon() throws IOException {
+    LOGGER.trace("Entering colon()");
+
+    if (currChar == ':') {
+      append();
+      getChar();
+      return makeToken(PreprocessorParser.DOUBLECOLON);
+    }
+    if (currIsSpace())
+      return makeToken(PreprocessorParser.LEXCOLON);
+    return makeToken(PreprocessorParser.OBJCOLON);
+  }
+
+  ProToken whitespace() throws IOException {
+    LOGGER.trace("Entering whitespace()");
+
+    loop : for (;;) {
+      switch (currChar) {
+        case ' ':
+        case '\t':
+        case '\f':
+        case '\n':
+        case '\r':
+          append();
+          getChar();
+          break;
+        default:
+          break loop;
+      }
+    }
+    return makeToken(PreprocessorParser.WS);
+  }
+
+  ProToken comment() throws IOException {
+    LOGGER.trace("Entering comment()");
+
+    // Escapes in comments are processed because you can end a comment
+    // with something dumb like: ~*~/
+    // We preserve that text.
+    // Note that macros are *not* expanded inside comments.
+    // (See the preprocessor source)
+    prepro.setDoingComment(true);
+    append(); // currChar=='*'
+    int commentLevel = 1;
+    while (commentLevel > 0) {
+      getChar();
+      unEscapedAppend();
+      if (currChar == '/') {
+        getChar();
+        unEscapedAppend();
+        if (currChar == '*')
+          commentLevel++;
+      } else if (currChar == '*') {
+        while (currChar == '*') {
+          getChar();
+          unEscapedAppend();
+          if (currChar == '/')
+            commentLevel--;
+        }
+      } else if (currInt == EOF_CHAR) {
+        prepro.lexicalThrow("Missing end of comment");
+      }
+    }
+    prepro.setDoingComment(false);
+    getChar();
+    return makeToken(PreprocessorParser.COMMENT);
+  }
+
+  ProToken singleLineComment() throws IOException {
+    LOGGER.trace("Entering singleLineComment()");
+
+    // Single line comments are treated just like regular comments,
+    // everything till end of line is considered comment - no escape
+    // character to look after
+    append(); // currChar=='/'
+
+    while (true) {
+      getChar();
+      unEscapedAppend();
+      if (currChar == '\r' || currChar == '\n' || currInt == EOF_CHAR) {
+        return makeToken(PreprocessorParser.COMMENT);
+      }
+    }
+  }
+
+  ProToken quotedString() throws IOException {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "quotedString" is 26 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)